library/std/src/sys/sync/rwlock/queue.rs

```diff
@@ -39,16 +39,16 @@
 //!
 //! ## State
 //!
-//! A single [`AtomicPtr`] is used as state variable. The lowest three bits are used
+//! A single [`AtomicPtr`] is used as state variable. The lowest four bits are used
 //! to indicate the meaning of the remaining bits:
 //!
-//! | [`LOCKED`] | [`QUEUED`] | [`QUEUE_LOCKED`] | Remaining    |                                                                                                                             |
-//! |:-----------|:-----------|:-----------------|:-------------|:----------------------------------------------------------------------------------------------------------------------------|
-//! | 0          | 0          | 0                | 0            | The lock is unlocked, no threads are waiting                                                                                |
-//! | 1          | 0          | 0                | 0            | The lock is write-locked, no threads waiting                                                                                |
-//! | 1          | 0          | 0                | n > 0        | The lock is read-locked with n readers                                                                                      |
-//! | 0          | 1          | *                | `*mut Node`  | The lock is unlocked, but some threads are waiting. Only writers may lock the lock                                          |
-//! | 1          | 1          | *                | `*mut Node`  | The lock is locked, but some threads are waiting. If the lock is read-locked, the last queue node contains the reader count |
+//! | [`DOWNGRADE`] | [`LOCKED`] | [`QUEUED`] | [`QUEUE_LOCKED`] | Remaining    |                                                                                                                             |
```

Comment:
Oh, I see this is done in the final commit. It could probably move to this one.

For reference, it's better to put cleanup commits first. Because obviously you only add perfect code that doesn't need cleanup :) so you're not adding code and then immediately undoing it. 

(At a pedantic level, commits adding higher-level functionality (public API) and tests should come after lower-level implementation commits, since you can't test something that isn't there yet (so bisecting to that point would be broken). But there are no requirements for this codebase, so don't worry too much about that).

Reply:
I've reordered the commits so the actual code changes should be easier to follow now